Makefile

```diff
@@ -161,22 +167,83 @@ build-dev: ## Build the project in development mode (Supports `ENVIRONMENT=true`
 
 build-all: build-x86_64-unknown-linux-musl build-aarch64-unknown-linux-musl ## Build the project in release mode for all supported platforms
 
-build-x86_64-unknown-linux-gnu: ## Build dynamically linked binary in release mode for the x86_64 architecture
-	$(RUN) build-x86_64-unknown-linux-gnu
+build-x86_64-unknown-linux-gnu: target/x86_64-unknown-linux-gnu/release/vector ## Build a release binary for the x86_64-unknown-linux-gnu triple.
+	@echo "Output to ${<}"
+
+build-aarch64-unknown-linux-gnu: target/aarch64-unknown-linux-gnu/release/vector ## Build a release binary for the aarch64-unknown-linux-gnu triple.
+	@echo "Output to ${<}"
 
 build-x86_64-unknown-linux-musl: ## Build static binary in release mode for the x86_64 architecture
 	$(RUN) build-x86_64-unknown-linux-musl
 
 build-aarch64-unknown-linux-musl: load-qemu-binfmt ## Build static binary in release mode for the aarch64 architecture
 	$(RUN) build-aarch64-unknown-linux-musl
 
+##@ Cross Compiling
+.PHONY: cross-enable
+cross-enable: cargo-install-cross
+
+.PHONY: CARGO_HANDLES_FRESHNESS
+CARGO_HANDLES_FRESHNESS:
+	${EMPTY}
+
+# This is basically a shorthand for folks.
+# `cross-anything-triple` will call `cross anything --target triple` with the right features.
+.PHONY: cross-%
+cross-%: export PAIR =$(subst -, ,$($(strip @):cross-%=%))
+cross-%: export COMMAND ?=$(word 1,${PAIR})
+cross-%: export TRIPLE ?=$(subst ${SPACE},-,$(wordlist 2,99,${PAIR}))
+cross-%: export PROFILE ?= release
+cross-%: export RUSTFLAGS += -C link-arg=-s
+cross-%: cargo-install-cross
+	cross ${COMMAND} \
+		$(if $(findstring release,$(PROFILE)),--release,) \
+		--target ${TRIPLE} \
+		--no-default-features \
+		--features target-${TRIPLE}
+
+target/%/vector: export PAIR =$(subst /, ,$(@:target/%/vector=%))
```

Comment:
TIL you can set target specific variables like this :clap:

Reply:
You can also do `?=` to only set if not set, and `override` to force a specific variable to be set. One note though, the envs you export here you can't use in the target dependencies below. :(